### ISSUE TITLE
Right click on empty space in extended dock could open settings menu

### ISSUE THREAD

**Issue by jhasse** [micheleg/dash-to-dock#336]
When right-clicking on the empty space in the extended dock it would be nice if it pops up the settings menu (the same one you get by right-clicking on the applications button).

**Comment by micheleg** [comment]
Hi, there is a pull request  implementing this: #300.

**Comment by franglais125** [comment]
Based on #300, I implemented this here: https://github.com/franglais125/dash-to-dock/tree/right_click_menu

**Comment by franglais125** [comment]
Two caveats:
- It perhaps makes sense to only make it available when in panel mode.
- For now, the code uses `Main.layoutManager.getWorkAreaForMonitor(0)` to set the coordinates. This uses `monitorIndex  = 0` for simplicity.

**Comment by micheleg** [comment]
I tried your branch. To me it is wrongly positioned , and the dock hide while trying to hover the popupmenu (I'm using intellihide and a bottom dock).

I guess that without the extended mode it automatically became insensitive?

**Comment by franglais125** [comment]
For now the code is only good enough for panel mode... I didn't test with intellihide.

Does it make sense to make it available without the panel mode? Hovering the mouse on the edge seems to be quite hard to achieve. In that case, the "SHow Applications" icon is much easier to target.

**Comment by micheleg** [comment]
I agree that it's probably better to provide only reasonable and consistent targets. However, the main advantage of this feature is to provide access to the settings if one disables the show-app button. Not sure what is better.

**Comment by franglais125** [comment]
Ahh, I see. Let me then implement for both cases, and then if we don't like we simply remove the non-panel mode.

**Comment by franglais125** [comment]
Ok, I fixed it for the case where the panel is not extended.

Two remaining things:
- When the panel is at the bottom, there is a small vertical offset. I have no idea where this comes from. Is it intentional in PopupMenu when the Side is `BOTTOM`? Also, do you see this too?
- I don't really know how to make it work more gracefully with intellihide. We would probably need to check if the menu is open before hiding the dock. Do you think it's worth adding something like that?

**Comment by franglais125** [comment]
Ok, I fixed the intellihide problem. Would you mind giving it a try again?

**Comment by micheleg** [comment]
Mmm, it crashed my shell with this error:

> mutter:ERROR:core/workspace.c:1016:meta_workspace_get_work_area_for_monitor: assertion failed: (which_monitor >= 0)

because I did not have any monitor explicitly set as the one to put the dock.

Yes I noticed the space, my guess is that that spacing is the top panel height, which need to be taken into account. It works correctly on the show-apps button.

**Comment by franglais125** [comment]
Oops, sorry for the crash! I pushed another commit to fix that.

As for the space, yes, I figured the same thing, but I couldn't manage to get rid of it... The Top position works fine too (i.e. there is no extra space).
